JsonPatch/PatchExtensions.cs

```diff
@@ -7,136 +7,157 @@
 
 namespace Json.Patch
 {
+	/// <summary>
+	/// Provides extension methods for <see cref="JsonPatch"/>.
+	/// </summary>
 	public static class PatchExtensions
 	{
 		/// <summary>
-		/// Apply patch to object
+		/// Apply a patch to an object to transform its data.
 		/// </summary>
-		/// <param name="patch">JsonPatch object</param>
-		/// <param name="obj">Object to patch</param>
-		/// <param name="options">Options</param>
-		/// <typeparam name="T">Any object</typeparam>
+		/// <param name="patch">The patch</param>
+		/// <param name="obj">The object</param>
+		/// <typeparam name="T">The type of the object.</typeparam>
 		/// <returns>New instance of patched object</returns>
-		public static T ApplyPatch<T>(this Json.Patch.JsonPatch patch, T obj, JsonSerializerOptions options = null)
+		/// <exception cref="InvalidOperationException">Thrown when the patch cannot be applied.</exception>
+		public static T Apply<T>(this JsonPatch patch, T obj)
 		{
-			using var doc = JsonDocument.Parse(JsonSerializer.Serialize(obj, options));
-			var res = patch.Apply(doc.RootElement).Result;
-			var result = JsonSerializer.Deserialize<T>(res.GetRawText(), options);
+			return Apply<T, T>(patch, obj);
+		}
+
+		/// <summary>
+		/// Apply a patch to an object to transform it to another type.
+		/// </summary>
+		/// <param name="patch">The patch</param>
+		/// <param name="obj">The object</param>
+		/// <typeparam name="TOriginal">The type of the original object.</typeparam>
+		/// <typeparam name="TTarget">The type of the target object.</typeparam>
+		/// <returns>New instance of patched object</returns>
+		/// <exception cref="InvalidOperationException">Thrown when the patch cannot be applied.</exception>
+		public static TTarget Apply<TOriginal, TTarget>(this JsonPatch patch, TOriginal obj)
+		{
+			using var doc = JsonDocument.Parse(JsonSerializer.Serialize(obj));
```

Comment:
Could you please add JsonSerializer options here? I think it's necessary in all serialization\deserialization code.

Reply:
This is putting submitting an object and returning an object.  The intermediate patch should make no difference.

Reply:
Sometimes there could be situation where object need special converter and it could be passed over options. Without it serialization could be impossible. For example in case of complex objects that are used in external library and there no defined converter for such model, but it could be used.
Also I mentions earlier, JsonPatch could be from web API which would be in camelCase, but default serialization would be in PascalCase and won't working.